services/src/test/java/org/keycloak/test/broker/oidc/AbstractOAuth2IdentityProviderTest.java

```diff
@@ -121,10 +132,54 @@ public void getFederatedIdentity_responseUrlLine() {
 		Assert.assertEquals("458rtf", fi.getId());
 	}
 
-	private TestProvider getTested() {
+	@Test
```

Comment:
I'm not following this test. It doesn't seem to be a proper functional test which is what we need. The tests should have a identity provider configured with the pass login hint option enabled. Then go to the login with the login_hint param, click on the link and check that the username field is prefilled in the external IdP login screen (external IdP can be another KC realm on the same server). We should also check that if the pass login_hint option is disabled the username field is not prefilled.